File: Java/Problem3.java
import java.util.HashMap;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int pointer = 0;
        int max_sub = 0;
        HashMap<Character, Integer> seen_chars = new HashMap<>();

        for(int index = 0; index < s.length(); index++){
            char current = s.charAt(index);

            if (seen_chars.containsKey(current) && seen_chars.get(current) >= pointer) {
                pointer = seen_chars.get(current) + 1;
            }

            seen_chars.put(current, index);
            max_sub = Math.max(max_sub, index - pointer + 1);
        }
        return max_sub;
    }
}

File: Python/Problem3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max_sub = 0
        pointer = 0
        seen_chars = {}
        for index in range(len(s)):
            if s[index] in seen_chars and seen_chars[s[index]] >= pointer:
                pointer = seen_chars[s[index]] + 1
            seen_chars[s[index]] = index
            max_sub = max(max_sub, index - pointer + 1)
        return max_sub
